### ISSUE TITLE
[BUG][Accessibility] Error summary links fail contrast on mouse hover

### ISSUE THREAD

**Issue by jamiegaskell** [scottish-government-design-system/design-system#129]
## Description of the issue

On mouse hover, error summary links have a contrast of 4.4:1. This fails WCAG 1.4.3 Contrast (minimum).

## Steps to reproduce the issue

1. Hover over the links
2. Obtain the colour hex values
3. Observe the contrast is only 4.4:1

## Expected behaviour

Contrast of text within interactive elements should be at least 4.5:1 on mouse hover and when they receive keyboard focus.

## Screenshots

![The red links have a contrast of 4.4:1 on mouse hover](https://github.com/user-attachments/assets/71f883ab-789d-45a9-adb7-372dcc57c579)

![DOM, the colours within the CSS](https://github.com/user-attachments/assets/ec69da26-7dfb-4f70-a115-4516ae989682)

**Comment by jsutcliffe** [comment]
Thanks for pointing that out. Those links should have been altered to use the new link contrast helper SASS functions but were missed. This will be fixed in the next DS release.

<img width="809" alt="Fixed contrast ratio on error summary links" src="https://github.com/user-attachments/assets/5b7e90bf-2c18-4c6e-b1f0-b3ee1d228201">

**Comment by jamiegaskell** [comment]
Apologies, upon digging deeper into this, I believe the text can be classified as large:

* The `font-size` is 19px
* The `font-weight` is 700

As the `font-weight` is 700, text larger than 18.5px is no longer regular sized text. The minimum required contrast then is 3:1.

<img width="779" alt="The computed font size is 19px and font weight 700" src="https://github.com/user-attachments/assets/0f61efd0-79b1-4e8c-aa5f-49d53f995f18">

Please disregard this issue

**Comment by jsutcliffe** [comment]
On small screens the error summary uses a text size of 16px, so it would still fail a contrast check. The visual appearance is also unsatisfactory and does not meet our expectation of what a link does when hovered (foreground colour should be a shade of the link text colour, background colour should be a tint of the link text colour).

**Comment by dkindnes** [comment]
This has been fixed in DS [v2.3.1](https://github.com/scottish-government-design-system/design-system/blob/main/CHANGELOG.md)

[Error summary](https://designsystem.gov.scot/components/error-summary )
